be/src/exec/vectorized/hdfs_scan_node.cpp

```diff
@@ -243,12 +246,13 @@ Status HdfsScanNode::_create_and_init_scanner(RuntimeState* state, const HdfsFil
     scanner_params.min_max_conjunct_ctxs = _min_max_conjunct_ctxs;
     scanner_params.min_max_tuple_desc = _min_max_tuple_desc;
     scanner_params.hive_column_names = &_hive_column_names;
-    scanner_params.parent = this;
     scanner_params.open_limit = hdfs_file_desc.open_limit;
+    scanner_params.profile = &_profile;
 
     HdfsScanner* scanner = nullptr;
     if (hdfs_file_desc.hdfs_file_format == THdfsFileFormat::PARQUET) {
```

Comment:
We could create a factory class for creating scanner, so `HdfsScanNode` doesn't have know anything about `HdfsParquetScanner` or `HdfsOrcScanner`.

Reply:
That works. But I don't see much benefit right now because there is only three formats. Maybe we can refactor that part later when more formats are supported. Laziness is a virtue